firebase_admin/db.py

```diff
@@ -0,0 +1,641 @@
+            return cls._type_object
+
+    @classmethod
+    def _extract_child(cls, value, path):
+        segments = path.split('/')
+        current = value
+        for segment in segments:
+            if isinstance(current, dict):
+                current = current.get(segment)
+            else:
+                return None
+        return current
+
+    def _compare(self, other):
+        """Compares two _SortEntry instances.
+
+        If the indices have the same numeric or string type, compare them directly. Ties are
+        broken by comparing the keys. If the indices have the same type, but are neither numeric
+        nor string, compare the keys. In all other cases compare based on the ordering provided
+        by index types.
+        """
+        self_key, other_key = self.index_type, other.index_type
+        if self_key == other_key:
+            if self_key in (self._type_numeric, self._type_string) and self.index != other.index:
+                self_key, other_key = self.index, other.index
+            else:
+                self_key, other_key = self.key, other.key
+
+        if self_key < other_key:
+            return -1
+        elif self_key > other_key:
+            return 1
+        else:
+            return 0
+
+    def __lt__(self, other):
+        return self._compare(other) < 0
+
+    def __le__(self, other):
+        return self._compare(other) <= 0
+
+    def __gt__(self, other):
+        return self._compare(other) > 0
+
+    def __ge__(self, other):
+        return self._compare(other) >= 0
+
+    def __eq__(self, other):
+        return self._compare(other) is 0
+
+
+class _Client(object):
+    """HTTP client used to make REST calls.
+
+    _Client maintains an HTTP session, and handles authenticating HTTP requests along with
+    marshalling and unmarshalling of JSON data.
+    """
+
+    def __init__(self, url=None, auth=None, session=None):
+        self._url = url
+        self._auth = auth
+        self._session = session
+
+    @classmethod
+    def from_app(cls, app):
+        """Created a new _Client for a given App"""
+        url = app.options.get('dbURL')
+        if not url or not isinstance(url, six.string_types):
+            raise ValueError(
+                'Invalid dbURL option: "{0}". dbURL must be a non-empty URL string.'.format(url))
+        parsed = urllib.parse.urlparse(url)
+        if parsed.scheme != 'https':
+            raise ValueError(
+                'Invalid dbURL option: "{0}". dbURL must be an HTTPS URL.'.format(url))
+        elif not parsed.netloc.endswith('.firebaseio.com'):
+            raise ValueError(
+                'Invalid dbURL option: "{0}". dbURL must be a valid URL to a Firebase realtime '
+                'database instance.'.format(url))
+        return _Client('https://{0}'.format(parsed.netloc), _OAuth(app), requests.Session())
+
+    def request(self, method, urlpath, **kwargs):
+        return self._do_request(method, urlpath, **kwargs).json()
```

Comment:
Some comments about what these do would be helpful, especially for what are valid values of `method` and what kinds of keyword arguments are taken here.